symphony-bdk-legacy/symphony-bdk-bot-sdk-java/src/test/java/com/symphony/bdk/bot/sdk/symphony/HealthCheckClientImplTest.java

```diff
@@ -0,0 +1,94 @@
+package com.symphony.bdk.bot.sdk.symphony;
+
+import static org.junit.Assert.assertEquals;
+
+import com.symphony.bdk.bot.sdk.lib.restclient.RestClient;
+import com.symphony.bdk.bot.sdk.lib.restclient.RestClientConnectionException;
+import com.symphony.bdk.bot.sdk.lib.restclient.RestClientImpl;
+import com.symphony.bdk.bot.sdk.symphony.model.HealthCheckInfo;
+
+import clients.SymBotClient;
+import configuration.SymConfig;
+import configuration.SymConfigLoader;
+import io.github.resilience4j.bulkhead.BulkheadConfig;
+import io.github.resilience4j.circuitbreaker.CircuitBreakerConfig;
+import model.HealthcheckResponse;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.web.client.RestTemplate;
+
+import java.time.Duration;
+
+public class HealthCheckClientImplTest {
+
+  private SymBotClient symBotClient;
+  private RestClient restClient;
+  private HealthcheckClientImpl healthcheckClientImpl;
+
+  @Value("${management.symphony-api-client-version}")
+  private String symphonyApiClientVersion;
+
+  @Before
+  public void initBot() {
```

Comment:
To actually test the functions, you should mock the following fields of `HealthcheckClientImpl`:
```
private SymBotClient symBotClient;
private RestClient restClient;
```
especially: 
`symBotClient.getConfig().getPodHost()`
`restClient.getRequest(getPodHealthUrl(), String.class);`
`symBotClient.getHealthcheckClient().performHealthCheck()`

Reply:
With my modifications, the coverage has increased : 
![coverage_184](https://user-images.githubusercontent.com/68892288/103543779-5c493400-4e9f-11eb-9a76-8d8ff7962b08.JPG)

Reply:
This is good ! You can go a step further with two other cases: when `this.restClient.getRequest` throws an exception and when `symBotClient.getHealthcheckClient().performHealthCheck()` throws an exception. This will increase the coverage.

Reply:
In my opinion it won't increase the coverage of `HealthCheckClientImpl` ...

Reply:
i'll try

Reply:
`performHealthCheck` is not a method of `HealthCheckClientImpl` .. How do you want me to do ?

Reply:
Moreover it's called in a private method. It's the same for `getRequest`

Reply:
Wait ..... I'm trying

Reply:
performHealthCheck doesn't throw any exception